lib/src/main/java/scal/io/liger/MainActivity.java

```diff
@@ -642,24 +679,48 @@ protected void onActivityResult(int requestCode, int resultCode, Intent intent)
                 }
 
                 // FIXME this can get a file:// uri, e.g. from facebook: https://rink.hockeyapp.net/manage/apps/30627/app_versions/62/crash_reasons/24334871
-                //String path = Utility.getRealPathFromURI(getApplicationContext(), uri);
+
+                // String path = Utility.getRealPathFromURI(getApplicationContext(), uri);
                 String path = FileUtils.getPath(getApplicationContext(), uri);
+
                 Log.d(TAG, "onActivityResult, imported file path:" + path);
                 String pathId = this.getSharedPreferences(Constants.PREFS_FILE, Context.MODE_PRIVATE).getString(Constants.PREFS_CALLING_CARD_ID, null); // FIXME should be done off the ui thread
 
                 Card c = mStoryPathLibrary.getCurrentStoryPath().getCardById(pathId);
 
                 if (c instanceof ClipCard) {
-                    ClipCard cc = (ClipCard)c;
+                    ClipCard cc = (ClipCard) c;
 
-                    MediaFile mf = new MediaFile(uri.toString(), cc.getMedium());
-                    cc.saveMediaFile(mf);
+                    String checkType = cc.getMedium();
+
+                    // adjust target value if needed
+                    if (checkType.equals(Constants.PHOTO)) {
+                        checkType = Constants.IMAGE;
+                    }
 
-                    // SEEMS LIKE A REASONABLE TIME TO SAVE
-                    mStoryPathLibrary.save(true);
+                    // confirm mime type
+                    String mimeType = "";
 
-                    mCardAdapter.changeCard(cc);
-                    scrollRecyclerViewToCard(cc);
+                    if (path != null) {
+                        mimeType = URLConnection.guessContentTypeFromName(path); // TODO: a null path causes failure, is that ok?
+                    }
+                    
+                    Log.d(TAG, "onActivityResult, media type is " + mimeType);
+
+                    if (mimeType.startsWith(checkType)) {
+
+                        MediaFile mf = new MediaFile(uri.toString(), cc.getMedium());
+                        cc.saveMediaFile(mf);
+
+                        // SEEMS LIKE A REASONABLE TIME TO SAVE
+                        mStoryPathLibrary.save(true);
+
+                        mCardAdapter.changeCard(cc);
+                        scrollRecyclerViewToCard(cc);
+                    } else {
+                        Log.e(TAG, "onActivityResult, expected " + checkType + " but found " + mimeType);
```

Comment:
instead of silently logging this, we need to raise the issue to the user so they know why their new import just vanished